public/script.js

```diff
@@ -1682,9 +1663,9 @@ async function Generate(type, automatic_trigger, force_name2) {
                         is_add_personality = true;
                         //chatString = chatString.substr(0,chatString.length-1);
                         //anchorAndPersonality = "[Genre: roleplay chat][Tone: very long messages with descriptions]";
-                        if ((anchorTop != "" || charPersonality != "") && !is_pygmalion) {
-                            if (anchorTop != "") charPersonality += ' ';
-                            item += "[" + charPersonality + anchorTop + ']\n';
+                        let personalityAndAnchor = [ charPersonality, anchorTop ].filter(x => x).join(' ');
+                        if (personalityAndAnchor && !is_pygmalion) {
+                            item += "[" + personalityAndAnchor + ']\n';
```

Comment:
Character without personality, before and after this PR:
```
DESCRIPTION
Then the roleplay chat between You and Test begins.
Test: Hello
[ Test Elaborate speaker]
You: Lorem ipsum
```
```
DESCRIPTION
Then the roleplay chat between You and Test begins.
Test: Hello
[Test Elaborate speaker]
You: Lorem ipsum
```

Character with personality, before and after this PR looks the same, so just for reference:
```
DESCRIPTION
Then the roleplay chat between You and Test begins.
Test: Hello
[Test's personality: PERSONALITY Test Elaborate speaker]
You: Lorem ipsum
```